pkg/theia/commands/policy_recommendation_delete.go

```diff
@@ -19,132 +19,78 @@ import (
-		}
-
-		clientset.CoreV1().RESTClient().Delete().
-			AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
-			Namespace(config.FlowVisibilityNS).
-			Resource("sparkapplications").
-			Name("pr-" + recoID).
-			Do(context.TODO())
-
-		err = deletePolicyRecommendationResult(clientset, kubeconfig, endpoint, useClusterIP, recoID)
-		if err != nil {
-			return err
-		}
-
-		fmt.Printf("Successfully deleted policy recommendation job with ID %s\n", recoID)
-		return nil
-	},
+	RunE: policyRecommendationDelete,
 }
 
-func getPolicyRecommendationIdMap(clientset kubernetes.Interface, kubeconfig string, endpoint string, useClusterIP bool) (idMap map[string]bool, err error) {
-	idMap = make(map[string]bool)
-	sparkApplicationList := &sparkv1.SparkApplicationList{}
-	err = clientset.CoreV1().RESTClient().Get().
-		AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
-		Namespace(config.FlowVisibilityNS).
-		Resource("sparkapplications").
-		Do(context.TODO()).Into(sparkApplicationList)
+func policyRecommendationDelete(cmd *cobra.Command, args []string) error {
+	prName, err := cmd.Flags().GetString("name")
 	if err != nil {
-		return idMap, err
+		return err
 	}
-	for _, sparkApplication := range sparkApplicationList.Items {
-		id := sparkApplication.ObjectMeta.Name[3:]
-		idMap[id] = true
+	if prName == "" && len(args) == 1 {
+		prName = args[0]
 	}
-	completedPolicyRecommendationList, err := getCompletedPolicyRecommendationList(clientset, kubeconfig, endpoint, useClusterIP)
+	err = ParseRecommendationName(prName)
 	if err != nil {
-		return idMap, err
+		return err
 	}
-	for _, completedPolicyRecommendation := range completedPolicyRecommendationList {
-		idMap[completedPolicyRecommendation.id] = true
+	useClusterIP, err := cmd.Flags().GetBool("use-cluster-ip")
+	if err != nil {
+		return err
 	}
-	return idMap, nil
-}
-
-func deletePolicyRecommendationResult(clientset kubernetes.Interface, kubeconfig string, endpoint string, useClusterIP bool, recoID string) (err error) {
-	connect, portForward, err := SetupClickHouseConnection(clientset, kubeconfig, endpoint, useClusterIP)
-	if portForward != nil {
-		defer portForward.Stop()
+	theiaClient, pf, err := SetupTheiaClientAndConnection(cmd, useClusterIP)
+	if err != nil {
+		return fmt.Errorf("couldn't setup theia client, %v", err)
 	}
+	if pf != nil {
+		defer pf.Stop()
+	}
+	err = theiaClient.Delete().
+		AbsPath("/apis/intelligence.theia.antrea.io/v1alpha1/").
+		Resource("networkpolicyrecommendations").
+		Name(prName).
+		Do(context.TODO()).
+		Error()
 	if err != nil {
-		return err
+		return fmt.Errorf("error when deleting policy recommendation job: %v", err)
 	}
-	query := "ALTER TABLE recommendations_local ON CLUSTER '{cluster}' DELETE WHERE id = (?);"
-	_, err = connect.Exec(query, recoID)
+	err = wait.Poll(config.DeletionPollInterval, config.DeletionPollTimeout, func() (bool, error) {
+		_, err = getPolicyRecommendationByName(theiaClient, prName)
+		if err == nil {
+			return false, nil
+		} else {
```

Comment:
do we want to explicitly check for the not found error for poll to return true? also, do we want to make the wait a optional param?

Reply:
Actually, I think we don't need this poll(). The api-server already propagate the error back to cli.